llvm/tools/llvm-exegesis/lib/RISCV/Target.cpp

```diff
@@ -131,9 +707,38 @@ class ExegesisRISCVTarget : public ExegesisTarget {
                                  MCOperand &AssignedValue,
                                  const BitVector &ForbiddenRegs) const override;
 
+  std::unique_ptr<SnippetGenerator> createSerialSnippetGenerator(
+      const LLVMState &State,
+      const SnippetGenerator::Options &Opts) const override {
+    return std::make_unique<RISCVSnippetGenerator<SerialSnippetGenerator>>(
+        State, Opts);
+  }
+
+  std::unique_ptr<SnippetGenerator> createParallelSnippetGenerator(
+      const LLVMState &State,
+      const SnippetGenerator::Options &Opts) const override {
+    return std::make_unique<RISCVSnippetGenerator<ParallelSnippetGenerator>>(
+        State, Opts);
+  }
+
   std::vector<InstructionTemplate>
   generateInstructionVariants(const Instruction &Instr,
                               unsigned MaxConfigsPerOpcode) const override;
+
+  void addTargetSpecificPasses(PassManagerBase &PM) const override {
+    // Turn AVL operand of physical registers into virtual registers.
+    PM.add(exegesis::createRISCVPreprocessingPass());
+    PM.add(createRISCVInsertVSETVLIPass());
```

Comment:
Does this require InsertVSETVLI to continue working on MIR without live intervals set up? @lukel97 do we plan to make VSETVLI insertion require live intervals?